opencl-source-provider.cmake

```diff
@@ -0,0 +1,50 @@
+set(BDA_DIR opm/simulators/linalg/bda)
+set(KERNELS_DIR ${BDA_DIR}/opencl/kernels)
+
+if(DEBUG_KERNELS)
+  set(DEBUG_DIR ${KERNELS_DIR}/.debug)
+
+  execute_process(
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEBUG_DIR}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+  )
+endif()
+
+set(CL_SRC_FILE ${BDA_DIR}/clSources.cpp)
```

Comment:
Please use `${PROJECT_BINARY_DIR}/${BDA_DIR}/clSources.cpp` to create this file in the build directory.